sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs

```diff
@@ -18,24 +19,23 @@
 
 namespace Microsoft.Azure.WebJobs.Host.EndToEndTests
 {
-    [Category("Live")]
-    public class EventHubEndToEndTests
+    public class EventHubEndToEndTests: LiveTestBase<EventHubsTestEnvironment>
```

Comment:
Please consider using the Event Hubs infrastructure to create a dynamic Event Hub and Storage scope in your test setup rather than a static resource.   [This snippet](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Snippets/Sample01_HelloWorldLiveTests.cs#L41) test is an example of what I'm advocating.

Reply:
Most other libraries create all resources they need as part of the ARM template. Is there a reason to create them dynamically?

Reply:
The local development story, both for ourselves and community contributors.    Events cannot be deleted from an Event Hub so the only way to start with a clean slate is to remove the resource and deploy a new one.   

As a team, we do not have a good story for automated "F5" experience that automates the cleanup and deployment of resources.  The use of ephemeral Event Hub instances scoped to an individual test run allows that automation without developers having to manually manage the test resource scripts.   

Storage falls into the same category if you're using the same Event Hub name, you'll have to clean out the container after every run to avoid accidentally having a checkpoint used (especially one that points to an event that doesn't exist if you recreated the hub).    If you're using a new Event Hub for each run then you either accumulate old data that may incur storage costs or you have to remember to clean out the container each time.   Here, again, the ephemeral resource allows developers to just run the tests and not manage resources.

Reply:
Depending on the test expectations as you keep making improvements, you're potentially going to run into the "well, I want to know exactly what events each test will see."  That means that either you're going to need an Event Hub specific to that test or you'll have to ensure that tests are not run in parallel, run in a specific order, and that you carefully manage the checkpoints and/or default starting position for each test

Reply:
Moved to use the dynamically created scopes.